compiler/rustc_passes/src/stability.rs

```diff
@@ -788,8 +788,24 @@ impl<'tcx> Visitor<'tcx> for Checker<'tcx> {
     fn visit_path(&mut self, path: &'tcx hir::Path<'tcx>, id: hir::HirId) {
         if let Some(def_id) = path.res.opt_def_id() {
             let method_span = path.segments.last().map(|s| s.ident.span);
-            self.tcx.check_stability(def_id, Some(id), path.span, method_span)
+            let item_is_allowed =
+                self.tcx.check_stability(def_id, Some(id), path.span, method_span);
+
+            if item_is_allowed {
```

Comment:
IIUC, we only report unstability in the path segments if the path itself is stable.  Could you add this to the comment?